javascript-examples/mpds_gifs.html

```diff
@@ -0,0 +1,56 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
+<head>
+<meta charset="utf-8" />
+</head>
+<body>
+
+<div style="position:fixed;bottom:0;left:50%;width:400px;margin-left:-200px;text-align:center;">
+    <h1><a href="javascript:document.location.reload()">Load more</a></h1>
+</div>
+
+<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/optimade/dist/index.js"></script>
+<script type="module">
+"use strict";
+
+function get_random_formula_anonymous(){
+
+    let formula = '';
+    const abcd = 'ABCD',
+        arity = [2, 3, 4][ Math.floor(Math.random() * 3) ];
+
+    for (let i = 0; i < arity; i++){
+        formula += abcd.charAt(i) + [1, 2, 3, 4, 5, 6, 7, 8][ Math.floor(Math.random() * 8) ];
```

Comment:
According to the OPTIMADE standard, elements should be sorted descending on their proportionality number for the chemical_formula_anonymous field.
A proportionality number of one should be omitted.
So BrCaCl<sub>3</sub>Na<sub>2</sub> should become "A3B2CD".
The code you use here will, however, also generate chemical_formula_anonymous strings with a different order, such as "A1B2C1D3". 
It would be nice if you could change this method, so it will only generate valid values for the chemical_formula_anonymous field.

Reply:
Thanks for pointing out. But I think it's minor thing. Let's merge it as it is now and later anyone will be able to improve. I might have some free time in January to get back to that as well.

Reply:
```suggestion
    for (let i = 0; i < arity; i++){
        numbers.push(["", 2, 3, 4, 5, 6, 7, 8][ Math.floor(Math.random() * 8) ]);
    }
    numbers.sort(function(a, b){return b-a}); // sort the numbers in descending order

    for (let i = 0; i < arity; i++){
        formula += abcd.charAt(i) + numbers[i];
    }
```

This fixed the issues I described. It should still be improved by making sure that the greatest common denominator is 1. mdps however also does not seem to reduce the proportion numbers so for now it still returns some structures.